Security release notes 202304.0 (#1796)
spryker/spryker-docs: docs/scos/user/intro-to-spryker/releases/release-notes/release-notes-202304.0/security-release-notes-202304.0.md

### Affected modules

  `spryker/propel`: 1.0.0-3.37.0

### How to get the fix

Update the affected module `spryker/propel` to version 3.38.0.
```bash
composer require spryker/propel:"~3.38.0"
composer show spryker/propel # Verify the version
```

## Stored Cross-Site Scripting (XSS) in Marketplace

An attacker can inject malicious payloads into the merchant profile page in the Merchant portal that's executed in the Back Office. This vulnerability lets an attacker execute code in the context of the Back Office user.

### Affected modules

* `spryker/merchant-profile-merchant-portal-gui`: 1.0.0-2.1.0
* `spryker/product-merchant-portal-gui`: 1.0.0-3.3.1

### How to get the fix

For the module `spryker/product-merchant-portal-gui`, depending on the version you are using, follow these steps:

* If your version of `spryker/product-merchant-portal-gui` is earlier than 2.0.0, update to version 1.7.1:
```bash
composer require spryker/product-merchant-portal-gui:"~1.7.1" spryker/gui:"~3.48.0" spryker/util-sanitize-xss:"~1.1.0"
composer show spryker/product-merchant-portal-gui # Verify the version
```

> * If your version of `spryker/product-merchant-portal-gui` is 2.0.0 up to and including 3.0.0, update to version 2.15.1:

Reviewer: "2.0.0 up to" - does it mean later than 2.0.0?

Author: That's the phrase devs want to use to make sure it's obvious that the version range includes 2.0.0 and 3.0.0.

Reviewer: maybe then "2.0.0 and later, including 3.0.0"? But it is still up to you to decide on the wording here

```

* If your version of `spryker/product-merchant-portal-gui` is later than 3.0.0, update to version 3.3.2:
```bash
composer require spryker/product-merchant-portal-gui:"~3.3.2" spryker/gui:"~3.48.0" spryker/util-sanitize-xss:"~1.1.0"
composer show spryker/product-merchant-portal-gui # Verify the version
```

For module `spryker/merchant-profile-merchant-portal-gui`, depending on the version you are using, follow these steps:

* If your version of `spryker/merchant-profile-merchant-portal-gui` is earlier than 2.0.0, update to version 1.8.0:
```bash
composer require spryker/merchant-profile-merchant-portal-gui:"1.8.0" spryker/gui:"~3.48.0" spryker/util-sanitize-xss:"~1.1.0"
composer show spryker/merchant-profile-merchant-portal-gui # Verify the version
```

* If your version of `spryker/merchant-profile-merchant-portal-gui` is later than 2.0.0, update to version 2.2.0:
```bash
composer require spryker/merchant-profile-merchant-portal-gui:"~2.2.0" spryker/gui:"~3.48.0" spryker/util-sanitize-xss:"~1.1.0"
composer show spryker/merchant-profile-merchant-portal-gui # Verify the version
```

## Self Cross-Site Scripting (XSS) in CMS

> Administrators can place a malicious payload in Placeholders, which can be executed in the editing and the view site, resulting in a XSS vulnerability.

Author: The phrase was copied from release notes. We should ask the creator.

> Administrators can place a malicious payload in Placeholders, which can be executed in the editing and the view site, resulting in a XSS vulnerability.

Reviewer 2: It's taken from here https://spryker.atlassian.net/browse/VRM-14 
It means editing and view mode when you are working in CMS.

    {% endinfo_block %}

   * If your version of `spryker/cms-block-gui` is later than 2.0.0 but earlier than 2.10.0, update it to version 2.10.0:
   ```bash
   composer require spryker/cms-block-gui:"~2.10.0"
   composer show spryker/cms-block-gui # Verify the version
   ```

3. Upgrade `spryker/cms-gui` module version to 5.11.0:
```bash
composer require spryker/cms-gui:"~5.11.0"
composer show spryker/cms-gui # Verify the version
```

4. Upgrade `spryker/merchant-profile-gui` module version to 1.1.0:
```bash
composer require spryker/merchant-profile-gui:"~1.1.0"
composer show spryker/merchant-profile-gui # Verify the version
```
5. Update `spryker/product-merchant-portal-gui`:

   * If your version of `spryker/product-merchant-portal-gui` is earlier than 2.0.0, update to version 1.7.0:
   ```bash
   composer require spryker/product-merchant-portal-gui:"~1.7.0"
   composer show spryker/product-merchant-portal-gui # Verify the version
   ```

>    * If your version of `spryker/product-merchant-portal-gui` is 2.0.0 up to and including 3.0.0, update to version 2.15.0:

Reviewer: same here with "up to"

   ```bash
   composer require spryker/product-merchant-portal-gui:"~2.15.0"
   composer show spryker/product-merchant-portal-gui # Verify the version
   ```

   * If your version of `spryker/product-merchant-portal-gui` is later than 3.0.0, update to version 3.2.0:
   ```bash
   composer require spryker/product-merchant-portal-gui:"~3.2.0"
   composer show spryker/product-merchant-portal-gui # Verify the version
   ```

6. Update `spryker/user-merchant-portal-gui`:
   * If your version of `spryker/user-merchant-portal-gui` is earlier than 2.0.0, update to version 1.11.0:
   ```bash
   composer require spryker/user-merchant-portal-gui:"~1.11.0"
   composer show spryker/user-merchant-portal-gui # Verify the version
   ```

   * If your version of `spryker/user-merchant-portal-gui` is later than 2.0.0, update to version 2.3.0:
   ```bash
   composer require spryker/user-merchant-portal-gui:"~2.3.0"
   composer show spryker/user-merchant-portal-gui # Verify the version
   ```

## Temporary Denial of Service (DoS) in Back Office

> An attacker can cause a temporary denial of service in the Back Office application by inserting very large data while generating vouchers in the merchandise page.

Reviewer: ```suggestion
An attacker can cause a temporary denial of service in the Back Office application by inserting very large data while generating vouchers on the _Merchandising_ page.
```

### Affected modules

* `spryker/discount`: <=9.29.0
* `spryker-feature/promotions-discounts`: <=202212.0
* `spryker-feature/marketplace-promotions-discounts`: <=202212.0

### How to get the fix

Update the affected module `spryker/discount` to version 9.30.0
```bash
composer require spryker/discount:"~9.30.0"
composer show spryker/discount # Verify the version
```

## User session remains valid after deletion in the Back Office

> The user's session remains valid, even if the user is deleted in the Back Office. This results in the deleted user getting an exception page with stack trace information while trying to log into the Front Office.

Reviewer: ```suggestion
The user's session remains valid, even if the user is deleted in the Back Office. This results in the deleted user getting an exception page with stack trace information while trying to log in to the Storefront.
```

### Affected modules

* `spryker/customer`: <7.49.0
* `spryker/customer-storage`: <1.0.0
* `spryker/oauth`: <2.8.0
* `spryker/oauth-customer-validation`: <1.0.0
* `spryker/oauth-extension`: <1.8.0
* `spryker-shop/customer-validation-page`: <1.0.0
* `spryker-feature/customer-account-management`: <202212.0

### How to get the fix

> To apply the fix, the customer needs to follow the below steps:

Reviewer: ```suggestion
To apply the fix, follow these steps:
```

> 1. Install or update required modules:

Reviewer: ```suggestion
1. Install or update the required modules:
```

     */
    protected function getSynchronizationDataPlugins(): array
    {
        return [
            new CustomerInvalidatedStorageSynchronizationDataPlugin(),
            ...
        ];
    }
}
```

## Outdated jQuery in use

In this case, an outdated version of jQuery was used (3.5.1). This version was released on 4 May, 2020 and affected by publicly known vulnerabilities.

### Changes 

Updated JQuery to the required version 3.6.3.

### Affected modules

* `spryker/gui`: < 3.49.0
* `spryker/util-sanitize-xss`: < 1.1.0

### How to get the fix

1. Update the affected module `spryker/gui` to version 3.49.0:
```bash
composer require spryker/gui:"~3.49.0"
composer show spryker/gui # Verify the version
```

2. Update the affected module `spryker/util-sanitize-xss` to version 1.1.0:
```bash
composer require spryker/util-sanitize-xss:"~1.1.0"
composer show spryker/util-sanitize-xss # Verify the version
```

3. In the `package.json` file, modify the version of the JQuery package to 3.6.3:
   1. Add the following line:
   ```js
   "jquery": "~3.6.3"
   ```

   2. In the same directory, run the following command:
   ```bash
   npm install
   ```

## Vulnerable version of PHP in use

> In PHP versions 7.4.*x* and earlier than 7.4.28, 8.0.*x* and earlier than 8.0.16, and 8.1.*x* and earlier than  8.1.3, when using filter functions with the `FILTER_VALIDATE_FLOAT` filter and minimum and maximum limits, if the filter fails, you may trigger the use of allocated memory after free, which can result it crashes, and potentially in overwrite of other memory chunks and RCE. This issue affects code that uses `FILTER_VALIDATE_FLOAT` with minimum and maximum limits.

Author: The same, we should ask the creator of release notes. 
@lapin-sergej maybe you can comment on this

> In PHP versions 7.4.*x* and earlier than 7.4.28, 8.0.*x* and earlier than 8.0.16, and 8.1.*x* and earlier than  8.1.3, when using filter functions with the `FILTER_VALIDATE_FLOAT` filter and minimum and maximum limits, if the filter fails, you may trigger the use of allocated memory after free, which can result it crashes, and potentially in overwrite of other memory chunks and RCE. This issue affects code that uses `FILTER_VALIDATE_FLOAT` with minimum and maximum limits.

Reviewer 2: It means after free memory. It's a typical error that devs can do in C/C++ language. The original text is taken from here: https://nvd.nist.gov/vuln/detail/CVE-2021-21708

> In PHP versions 7.4.*x* and earlier than 7.4.28, 8.0.*x* and earlier than 8.0.16, and 8.1.*x* and earlier than  8.1.3, when using filter functions with the `FILTER_VALIDATE_FLOAT` filter and minimum and maximum limits, if the filter fails, you may trigger the use of allocated memory after free, which can result it crashes, and potentially in overwrite of other memory chunks and RCE. This issue affects code that uses `FILTER_VALIDATE_FLOAT` with minimum and maximum limits.

Author: ```suggestion
In PHP versions 7.4.*x* and earlier than 7.4.28, 8.0.*x* and earlier than 8.0.16, and 8.1.*x* and earlier than  8.1.3, when using filter functions with the `FILTER_VALIDATE_FLOAT` filter and minimum and maximum limits, if the filter fails, you may trigger the use of allocated memory after free, which can result in crashes, and potentially in the overwrite of other memory chunks and RCE. This issue affects code that uses `FILTER_VALIDATE_FLOAT` with minimum and maximum limits.
```

> In PHP versions 7.4.*x* and earlier than 7.4.28, 8.0.*x* and earlier than 8.0.16, and 8.1.*x* and earlier than  8.1.3, when using filter functions with the `FILTER_VALIDATE_FLOAT` filter and minimum and maximum limits, if the filter fails, you may trigger the use of allocated memory after free, which can result it crashes, and potentially in overwrite of other memory chunks and RCE. This issue affects code that uses `FILTER_VALIDATE_FLOAT` with minimum and maximum limits.

Reviewer: I'd suggest: In PHP versions 7.4.*x* and earlier than 7.4.28, 8.0.*x* and earlier than 8.0.16, 8.1.*x* and earlier than  8.1.3, when using filter functions with the `FILTER_VALIDATE_FLOAT` filter and minimum and maximum limits, if the filter fails, you may trigger the use of allocated memory after free memory. This can result in crashes, and potentially in overwriting of other memory chunks and RCE. This issue affects code that uses `FILTER_VALIDATE_FLOAT` with minimum and maximum limits.
